azurerm/internal/services/policy/policy_assignment_resource.go

```diff
@@ -251,6 +258,7 @@ func resourceArmPolicyAssignmentRead(d *schema.ResourceData, meta interface{}) e
 		d.Set("policy_definition_id", props.PolicyDefinitionID)
 		d.Set("description", props.Description)
 		d.Set("display_name", props.DisplayName)
+		d.Set("enforcement_mode", props.EnforcementMode)
```

Comment:
We'll also want to convert this into a boolean with something like:
```suggestion
		d.Set("enforcement_mode", props.EnforcementMode == policy.Default)
```

Reply:
I fixed it. Still not sure what this code actually does, or how this fix changes it. The tfstate was already correct and the code was correctly comparing state against definition.